tools/UpdatePaths/readme.md

```diff
@@ -1,32 +1,35 @@
+<h1>UpdatePaths</h1>
 
-Use this tool before using mapmerge or opening the map in an editor.
+<h2>HOW TO USE:</h2>
 
-IMPORTANT:
-Please tie the script you are making to the associated PR on github and put it in the scripts folder when you are done.
+Drag one of the scripts in the “Scripts” folder onto the .bat file “Update Paths” to open it with the `.bat` file (or use the Python script directly depending on your operating system). Let the script run to completion.
 
-For example: 67107_Turf_Updates_2
+Use this tool before using MapMerge2 or opening the map in an map editor. This is because the map editor may discard any unknown paths not found in the `.DME` file.
 
-HOW TO MAKE A SCRIPT:
-This tool updates paths in the game to new paths. For instance:
-If you have a path labeled
+---
 
-/obj/structure/door/airlock/science/closed/rd
+<h2>HOW TO MAKE A SCRIPT:</h2>
 
-and wanted it to be
+This tool updates paths in the game to new paths. For instance:
 
-/obj/structure/door/airlock/science/rd/closed
+If you have a path labeled `/obj/structure/door/airlock/science/closed/rd` and wanted it to be `/obj/structure/door/airlock/science/rd/closed`, this tool would update it for you! This is extremely helpful if you want to be nice to people who have to resolve merge conflicts from the PRs that you make updating these areas.
 
-This tool would update it for you! This is extremely helpful if you want to be nice to people who have to resolve merge conflicts from the PRs that you make updating these areas.
+<h3>How do you do it?</h3>
 
-How do you do it?
-Simply open a notepad and type this on a line:
+Simply create a `.TXT` file and type this on a line:
 
-/obj/structure/door/airlock/science/closed/rd : /obj/structure/door/airlock/science/rd/closed
+/obj/structure/door/airlock/science/closed/rd : /obj/structure/door/airlock/science/rd/closed{@OLD}
 
 The path on the left is the old, the path on the right is the new. It is seperated by a ":"
 If you want to make multiple path changes in one script, simply add more changes on new lines.
 
+Putting `{@OLD}` is important since otherwise, UpdatePaths will automatically discard the old variables attached to the old path. Adding `{@OLD}` to the right-hand side will ensure that every single variable from the old path will be applied to the new path.
+
+---
+
+<h3>On Variable Editing</h3>
+
+If you do not want any variable edits to carry over, you can simply skip adding the `{@OLD}` tag (although this is not advisable under normal circumstances). There are also a bunch of neat features you can use with UpdatePaths variable filtering, such as ensuring all new paths get a certain variable edit, filtering old paths if they have a certain variable edit, or even splitting one path into multiple paths on a map. You can find out more about this by reading `https://github.com/tgstation/tgstation/blob/master/tools/UpdatePaths/__main__.py#L9`.
```

Comment:
I had to put the link in a stupid code bracket because MD wouldn't read it as a link and apply formatting to `__main__` to make it look like _main_. If only we had such foresight back then.

Reply:
san what exactly do you want this to do?

Reply:
I think I was actually stupid and the markdown viewer I was using on my local machine was lying to me gimme a second

Reply:
![image](https://user-images.githubusercontent.com/34697715/179390500-3c5594a3-c9af-48db-bb9b-8a6f02aed4ce.png)

This is what the link looked like on my local preview. However, GitHub Parses it as:

![image](https://user-images.githubusercontent.com/34697715/179390537-7fb10263-023d-47f3-8ea5-132b4831e96e.png)

So we're fine actually. Never mind.